state/virtualhostkeys.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/mgo/v3"
+	"github.com/juju/mgo/v3/txn"
+)
+
+// unitHostKeyID provides the virtual host key
+// lookup value for a unit based on the unit name.
+func unitHostKeyID(unitName string) string {
+	return "unit" + "-" + unitName + "-" + "hostkey"
+}
+
+// machineHostKeyID provides the virtual host key
+// lookup value for a machine based on the machine ID.
+func machineHostKeyID(machineID string) string {
+	return "machine" + "-" + machineID + "-" + "hostkey"
+}
+
+// VirtualHostKey represents the state of a virtual host key.
+type VirtualHostKey struct {
+	st  *State
```

Comment:
This struct does not need an embedded state. 1. it's not used anyway in this PR, 2. that pattern is only needed for primary entities like machine, unit, application where methods on those entities actually read/write state. VirtualHostKey is just a POGO with a bit of date. Use the doc struct for bson serialisation, but VirtualHostKey as a struct probably isn't even needed as the only method on it is to get the host key content, so methods on state which need to return the host key for a unit or machine can just do so directly.